app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -76,4 +74,10 @@ def checkout_git_repository(target_directory)
     git_repository.update_repo
     git_repository.checkout(scm_branch, target_directory)
   end
+
+  private
+
+  def ensure_git_repository
+    self.git_repository_id ||= GitRepository.create!(:url => scm_url).id
```

Comment:
Does this update the record in place? Or does it work because this is a `before_validation` hook?